util/dnsname/dnsname_test.go

```diff
@@ -26,3 +79,22 @@ func TestHasSuffix(t *testing.T) {
 		}
 	}
 }
+
+func TestTrimSuffix(t *testing.T) {
+	tests := []struct {
+		name   string
+		suffix string
+		want   string
+	}{
+		{"foo.magicdnssuffix.", "magicdnssuffix", "foo"},
+		{"foo.magicdnssuffix", "magicdnssuffix", "foo"},
```

Comment:
add a case with ".magicdnssuffix" as suffix?

Reply:
I also updated https://github.com/tailscale/tailscale/pull/1304/files#diff-74c7bebcdbd6a71ad364ffc195be09d491ce09e0dc685fdc7ba4a462f5553f98R95 so that this would perform as expected.

`TrimSuffix("foo.magicdnssuffix",".magicdnssuffix") == "foo"`